File: test_soal2.php
<?php
$input1 = "hallo jesika24 selamat datang!";
$input2 = "hallo anggun selamat datang!";
$input3 = "hallo ** selamat datang!";
$input4 = "hallo Mariage889120! selamat datang!";
echo "inputan = ".$input1."<br>";
validation($input1);
echo "<br>";
echo "<br>";

echo "inputan = ".$input2."<br>";
validation($input2);
echo "<br>";
echo "<br>";

echo "inputan = ".$input3."<br>";
validation($input3);
echo "<br>";
echo "<br>";

echo "inputan = ".$input4."<br>";
validation($input4);
echo "<br>";


function validation ($string){
if (preg_match('#[0-9]#',$string)){
    $string = preg_replace("/[^0-9]/","",$string);
    echo "Sistem kami menolak untuk inputan berisi angka ".$string;
}
else{
    echo "Sistem memeriksa data anda valid dengan ketentuan kami dengan inputan ".$string;
}
}

?>

File: test_soal2.java
public class test_soal2 {
    public static void main(String[] args) {
        String input1 = "hallo jesika24 selamat datang!";
        String input2 = "hallo anggun selamat datang!";
        String input3 = "hallo ** selamat datang!";
        String input4 = "hallo Mariage889120! selamat datang!";
        
        userNameValid(input1);
        userNameValid(input2);
        userNameValid(input3);
        userNameValid(input4);
    }
    public static void userNameValid(String message) {
        String[] parts=message.split(" ");
        String userName = parts[1];
        String errNum="";
        for(int i=0;i<userName.length();i++){
            int check = userName.charAt(i);
            if(check>=48 && check<=57) errNum+=userName.charAt(i);
        }
        if(errNum==""){
            System.out.println("Sistem memeriksa data anda valid dengan ketentuan kami dengan inputan "+message);
        } else {
            System.out.println("Sistem kami menolak untuk inputan berisi angka "+errNum);
        }
    }
}
